include/object/notification.h

```diff
@@ -27,8 +27,12 @@ static inline void maybeReturnSchedContext(notification_t *ntfnPtr, tcb_t *tcb)
     if (sc == tcb->tcbSchedContext) {
         tcb->tcbSchedContext = NULL;
         sc->scTcb = NULL;
+        /* If the current thread returns its sched context then it should not
+           by default continue running. */
+        if (tcb == NODE_STATE(ksCurThread)) {
```

Comment:
I suspect this function is only ever called with the current thread, but this fix seems reasonable for now.

Reply:
It would be nice if this were checked *before* being encoded into the proofs.

Reply:
@pingerino It's not yet encoded into the proofs. I intentionally wanted to get systems approval before I did that. As of this afternoon I have a pull-request up on l4v, but I can pull that back if there is any uncertainty here.

Reply:
Ah, I misunderstood your comment. I understand now. Yes, we could have checked and changed this before we merged.